libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSessionLocationEngine.kt

```diff
@@ -55,6 +56,16 @@ internal class TripSessionLocationEngine constructor(
     private var locationEngineCallback = object : LocationEngineCallback<LocationEngineResult> {
         override fun onSuccess(result: LocationEngineResult?) {
             result?.locations?.lastOrNull()?.let {
+                // log if location is not fresh enough
+                val currentTime = SystemClock.elapsedRealtimeNanos()
+                val locationAge = (currentTime - it.elapsedRealtimeNanos)
+                if (locationAge > DELAYED_LOCATION_WARNING_THRESHOLD_NS) {
+                    logD(
+                        TAG,
+                        "Got obsolete location: current time = $currentTime, " +
+                            "elapsedRealtimeNanos = ${it.elapsedRealtimeNanos}"
+                    )
+                }
```

Comment:
maybe replace comment by a function:
```kotlin
            fun logIfLocationIsNotFreashEnough(location: Location) {
                val currentTime = SystemClock.elapsedRealtimeNanos()
                val locationAge = (currentTime - it.elapsedRealtimeNanos)
                if (locationAge > DELAYED_LOCATION_WARNING_THRESHOLD_NS) {
                    logD(
                        TAG,
                        "Got obsolete location: current time = $currentTime, " +
                            "elapsedRealtimeNanos = ${it.elapsedRealtimeNanos}"
                    )
                }
            }
```
and 
```suggestion
               logIfLocationIsNotFreashEnough(it)
```